mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/logging/impl/Log.java

```diff
@@ -101,4 +102,28 @@
 
 	@Message(id = 1228, value = "Query must be executed using 'getResultList()' or 'getSingleResult()' method: %s")
 	HibernateException readQueryMustBeExecutedViaGetResultList(MongoDBQueryDescriptor queryDescriptor);
+
+	@Message(id = 1229, value = "Unable to create index %2$s on collection %1$s")
+	HibernateException unableToCreateIndex(String collection, String indexName, @Cause Exception e);
+
+	@LogMessage(level = ERROR)
+	@Message(id = 1230, value = "Unable to create text index %2$s on collection %1$s. A text index named %3$s already exists and MongoDB only supports one text index per collection.")
+	void unableToCreateTextIndex(String collection, String newIndexName, String existingIndexName);
```

Comment:
From the exception it seems that the index already exists in the DB, so it's not that it is not created. In Neo4jSchemaDefiner we use the property: hibernate.schema_update.unique_constraint_strategy to let the user decide what to do in this situations.